[#41681] Improve the safety of remote upgrades

magenta-af opened the pull request:
This PR adds support for hooking into `jobmanager` execution and running extra scripts before and after it.

It also includes two scripts that make use of this functionality: one of them uses `nmcli` to forcibly activate network connections before checking in, and the other protects the configuration files against any changes.

ale-f committed:
[#41681] Add support for jobmanager hooks

agger-magenta commented on admin_scripts/hooks/network_up.sh:
```diff
@@ -0,0 +1,67 @@
+
+HOOKS="/etc/os2borgerpc/pre-checkin.d/network_up.sh"
+
+if [ "$1" != "" -a "$1" != "false" -a "$1" != "falsk" ]; then
```
Hvad med at ændre protokollen her lidt, så man også kan deaktivere ved at skrive "no" eller "nej"?

agger-magenta commented on admin_scripts/hooks/protect_config_files.sh:
```diff
@@ -0,0 +1,67 @@
+HOOKS="/etc/os2borgerpc/pre-checkin.d/protect_config_files.sh
+/etc/os2borgerpc/post-checkin.d/protect_config_files.sh"
+
+if [ "$1" != "" -a "$1" != "false" -a "$1" != "falsk" ]; then
```
Som ovenfor.

agger-magenta replied:
Det ser fint og helt rigtigt ud bortset fra min lille kommentar om ja/nej/true/false-protokollen for aktivering.

ale-f committed:
[#41681] Create hook folders if they're missing

I might have preferred to do this with curly bracket expansion, but that
turns out to be a bashism. I had no idea!
